atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/WriteOnceDeleteOnceWorkflows.java

```diff
@@ -0,0 +1,106 @@
+import com.palantir.atlasdb.buggify.impl.DefaultNativeSamplingSecureRandomFactory;
+import com.palantir.atlasdb.workload.store.ImmutableWorkloadCell;
+import com.palantir.atlasdb.workload.store.InteractiveTransactionStore;
+import com.palantir.atlasdb.workload.store.WorkloadCell;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import java.security.SecureRandom;
+import java.util.Optional;
+import java.util.function.IntFunction;
+
+/**
+ * The idea of this class is to catch bugs around tombstone drops, by having a
+ * workflow which _mostly_ writes once then deletes once for a given cell. Probability and randomness are used to
+ * enforce this behaviour, thus it's possible to have a write without a delete, or visa versa; some cells may
+ * randomly be written or deleted multiple times as well.
+ */
+public final class WriteOnceDeleteOnceWorkflows {
+
+    private static final SecureRandom RANDOM = DefaultNativeSamplingSecureRandomFactory.INSTANCE.create();
+
+    private static final Integer COLUMN = 0;
+
+    private WriteOnceDeleteOnceWorkflows() {
+        // Utility class
+    }
+
+    public static Workflow create(
+            InteractiveTransactionStore store,
+            WriteOnceDeleteOnceWorkflowConfiguration configuration,
+            ListeningExecutorService executionExecutor) {
+        return create(store, configuration, executionExecutor, DefaultBuggifyFactory.INSTANCE, RANDOM::nextInt);
+    }
+
+    @VisibleForTesting
+    static Workflow create(
+            InteractiveTransactionStore store,
+            WriteOnceDeleteOnceWorkflowConfiguration configuration,
+            ListeningExecutorService executionExecutor,
+            BuggifyFactory buggifyFactory,
+            IntFunction<Integer> randomNumberGenerator) {
+        WriteOnceDeleteOnceWorkflow workflow =
+                new WriteOnceDeleteOnceWorkflow(buggifyFactory, randomNumberGenerator, configuration);
+        return DefaultWorkflow.create(store, workflow::run, configuration, executionExecutor);
+    }
+
+    private static class WriteOnceDeleteOnceWorkflow {
+        private final BuggifyFactory buggifyFactory;
+        private final IntFunction<Integer> rowKeyGenerator;
+        private final WriteOnceDeleteOnceWorkflowConfiguration configuration;
+
+        public WriteOnceDeleteOnceWorkflow(
+                BuggifyFactory buggifyFactory,
+                IntFunction<Integer> rowKeyGenerator,
+                WriteOnceDeleteOnceWorkflowConfiguration configuration) {
+            this.buggifyFactory = buggifyFactory;
+            this.rowKeyGenerator = rowKeyGenerator;
+            this.configuration = configuration;
+        }
+
+        public Optional<WitnessedTransaction> run(InteractiveTransactionStore store, int taskIndex) {
+            configuration.transactionRateLimiter().acquire();
+            return store.readWrite(txn -> {
+                Integer rowKey = rowKeyGenerator.apply(configuration.maxKey());
```

Comment:
maybe not actionable: I think this works out fine in the end, but be aware that `maxKey` is never actually generated because of exclusivity so the name is potentially misleading

Reply:
probably a comment should suffice

Reply:
Added it to the configuration as a comment!